std/range/package.d

```diff
@@ -7809,6 +7809,31 @@ auto slide(Flag!"withPartial" f = Yes.withPartial,
     assert(d == ["AG"d: 2, "GA"d: 2]);
 }
 
+/// withPartial only has an effect if last element in the range doesn't have the full size
+@safe pure nothrow unittest
+{
+    import std.algorithm.comparison : equal;
+    import std.algorithm.iteration : map;
+
+    auto withPartial = iota(5, 10).map!(i => i.iota.slide!(Yes.withPartial)(3, 4));
```

Comment:
Can you simplify this to sliding over an `iota` rather than a `map` of an `iota`? It would make what's going on a lot more clear.